Sources_v3/StreamChat/Controllers/MessageController/MessageController_Tests.swift

```diff
@@ -1044,6 +1044,66 @@ final class MessageController_Tests: StressTestCase {
         // Assert controller is kept alive.
         AssertAsync.staysTrue(weakController != nil)
     }
+
+    // MARK: - Dispatch ephemeral message action
+
+    func test_dispatchEphemeralMessageAction_propagatesError() {
+        // Simulate `dispatchEphemeralMessageAction` call and catch the completion.
+        var completionError: Error?
+        controller.dispatchEphemeralMessageAction(.unique) { [callbackQueueID] in
+            AssertTestQueue(withId: callbackQueueID)
+            completionError = $0
+        }
+
+        // Simulate network response with the error.
+        let updaterError = TestError()
+        env.messageUpdater.dispatchEphemeralMessageAction_completion!(updaterError)
+
+        // Assert error is propagated.
+        AssertAsync.willBeEqual(completionError as? TestError, updaterError)
+    }
+
+    func test_dispatchEphemeralMessageAction_propagatesNilError() {
+        // Simulate `dispatchEphemeralMessageAction` call and catch the completion.
+        var completionIsCalled = false
+        controller.dispatchEphemeralMessageAction(.unique) { [callbackQueueID] error in
+            // Assert callback queue is correct.
+            AssertTestQueue(withId: callbackQueueID)
+            // Assert there is no error.
+            XCTAssertNil(error)
+            completionIsCalled = true
+        }
+
+        // Simulate successful updater call.
+        env.messageUpdater.dispatchEphemeralMessageAction_completion!(nil)
+
+        // Assert completion is called.
+        AssertAsync.willBeTrue(completionIsCalled)
+    }
+
+    func test_dispatchEphemeralMessageAction_callsUpdater_withCorrectValues() {
+        let action: AttachmentAction = .unique
+
+        // Simulate `dispatchEphemeralMessageAction` call.
+        controller.dispatchEphemeralMessageAction(action)
+
+        // Assert updater is called with correct values.
+        XCTAssertEqual(env.messageUpdater.dispatchEphemeralMessageAction_cid, controller.cid)
+        XCTAssertEqual(env.messageUpdater.dispatchEphemeralMessageAction_messageId, controller.messageId)
+        XCTAssertEqual(env.messageUpdater.dispatchEphemeralMessageAction_action, action)
+    }
+
+    func test_dispatchEphemeralMessageAction_keepsControllerAlive() {
+        // Simulate `dispatchEphemeralMessageAction` call.
+        controller.dispatchEphemeralMessageAction(.unique)
+
+        // Create a weak ref and release a controller.
+        weak var weakController = controller
+        controller = nil
+
+        // Assert controller is kept alive.
+        AssertAsync.staysTrue(weakController != nil)
+    }
```

Comment:
nit: We don't need to have an explicit test for this. Please refer to other tests testing this.

Reply:
Checked other tests in `MessageController`:
- for some methods we have `test_method_keepsControllerAlive`
- for others we check this in `test_method_propagatesNilError`

Frankly speaking, I find `test_method_keepsControllerAlive` approach more convenient: if someone decide to add `[weak self]` from failing `test_method_keepsControllerAlive` it's clear what is broken. 

When `test_method_propagatesNilError` becomes red it's not obvious.